proto/ibc/lightclients/wasm/v1/tx.proto

```diff
@@ -27,3 +30,19 @@ message MsgStoreCodeResponse {
   // the sha256 hash of the stored code
   bytes checksum = 1;
 }
+
+// MsgMigrateContract defines the request type for the MigrateContract rpc.
+message MsgMigrateContract {
+  option (cosmos.msg.v1.signer) = "signer";
+
+  string signer = 1;
+  // the client id of the contract
+  string client_id = 2;
+  // the code hash of the new wasm byte code for the contract
+  bytes code_hash = 3;
```

Comment:
Should this be `bytes` or hex encoded string? I wasn't sure which is more appropriate.

Reply:
We use `bytes` in `ClientState`, so we could also use `bytes` here.

Reply:
But the code hash query results are returned in a hex encoded string

Reply:
it seems like when we actually use the codehash we typically want a `string` format so my vote would be for string, otherwise the user has to convert from bytes to string every time anyway

Reply:
Can we get extra opinions here @colin-axner @damiannolan

Reply:
The field for the old code hash and the new code hash should probably be of the same type. Generally it is nice if the users don't need to do any conversions between the type they query for and the type they submit as input. The hex encoding is simply for readability? Making this a hex encoded string doesn't solve user issues when creating the client and needing to provide the initial code hash in bytes

Reply:
In general `x/wasm` uses `bytes` for the code hash (see [here](https://github.com/CosmWasm/wasmd/blob/a2bbd36364a78867319ad0a28f85203c8cdfbc8d/proto/cosmwasm/wasm/v1/tx.proto#L108) and [here](https://github.com/CosmWasm/wasmd/blob/a2bbd36364a78867319ad0a28f85203c8cdfbc8d/proto/cosmwasm/wasm/v1/tx.proto#L412)).